Src/Support/Google.Apis.Auth/GoogleJsonWebSignature.cs

```diff
@@ -14,15 +14,191 @@
+            using (var hashAlg = SHA256.Create())
+            {
+                hash = hashAlg.ComputeHash(Encoding.ASCII.GetBytes($"{parts[0]}.{parts[1]}"));
+            }
+            bool verifiedOk = false;
+            foreach (var googleCert in await GetGoogleCertsAsync(clock, forceGoogleCertRefresh, certsJson))
+            {
+#if NET45
+                verifiedOk = ((RSACryptoServiceProvider)googleCert).VerifyHash(hash, Sha256Oid, signature);
+#elif NETSTANDARD1_3
+                verifiedOk = googleCert.VerifyHash(hash, signature, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+#else
+#error Unsupported platform
+#endif
+                if (verifiedOk)
+                {
+                    break;
+                }
+            }
+            if (!verifiedOk)
+            {
+                throw new InvalidJwtException("JWT invalid: unable to verify signature.");
+            }
+
+            // Verify iss, iat and exp claims
+            if (payload.Issuer != "https://accounts.google.com" && payload.Issuer != "accounts.google.com")
+            {
+                throw new InvalidJwtException("JWT issuer incorrect. Must be 'https://accounts.google.com' or 'accounts.google.com'");
+            }
+            if (payload.IssuedAtTimeSeconds == null || payload.ExpirationTimeSeconds == null)
+            {
+                throw new InvalidJwtException("JWT must contain 'iat' and 'exp' claims");
+            }
+            var now = (clock.UtcNow - new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc)).TotalSeconds;
+            if (now < payload.IssuedAtTimeSeconds.Value || now > payload.ExpirationTimeSeconds.Value)
+            {
+                throw new InvalidJwtException("JWT is not yet valid, or has expired.");
+            }
+
+            // All verification passed, return payload.
+            return payload;
+        }
+
+        private static string Base64UrlToJson(string base64Url) => Encoding.UTF8.GetString(Base64UrlDecode(base64Url));
+
+        private static byte[] Base64UrlDecode(string base64Url)
+        {
+            var base64 = base64Url.Replace('-', '+').Replace('_', '/');
+            switch (base64.Length % 4)
+            {
+                case 2: base64 += "=="; break;
+                case 3: base64 += "="; break;
+            }
+            return Convert.FromBase64String(base64);
+        }
+
+        private static SemaphoreSlim _certCacheLock = new SemaphoreSlim(1);
+        private static DateTime _certCacheDownloadTime;
+        private static List<RSA> _certCache;
+
+        // internal for testing
+        internal static async Task<List<RSA>> GetGoogleCertsAsync(IClock clock, bool forceGoogleCertRefresh, string certsJson)
+        {
+            var now = clock.UtcNow;
+            await _certCacheLock.WaitAsync();
+            try
+            {
+                if (forceGoogleCertRefresh ||_certCache == null || _certCacheDownloadTime + CertCacheRefreshInterval < now)
```

Comment:
Space between || and `_certCache`.

Parenthesis arround `(_certCacheDownloadTime + CertCacheRefreshInterval)`.